camunda-formio-tasklist-vue/src/components/TaskList.vue

```diff
@@ -801,40 +764,88 @@ export default class Tasklist extends Mixins(TaskListMixin) {
     max: number
   ) {
     this.selectedfilterId = filterId;
-    await CamundaRest.filterTaskListPagination(
+    const paginatedTaskResults = await CamundaRest.filterTaskListPagination(
       this.token,
       filterId,
       requestData,
       first,
       max,
       this.bpmApiUrl
-    ).then((result) => {
-      this.tasks = result.data;
-    });
+    );
+    this.tasks = paginatedTaskResults.data;
+  }
+
+  async onUserSearch (search: string, loading: any) {
+    if(search.length) {
+      loading(true);
+      this.reviewerUsersList = [];
+    }
+
+    if(this.selectSearchType === SEARCH_OPTION_TYPE.FIRST_NAME) {
+      const firstNameUserList = await CamundaRest.getUsersByFirstNameGroups(this.token, this.bpmApiUrl, search, reviewerGroup);
+      if(firstNameUserList) {
+        this.reviewerUsersList = [];
+        firstNameUserList.data.forEach((user: UserPayload) => {
+          this.reviewerUsersList.push({
+            code: user.id,
+            email: user.email!,
+            firstName: `${user.firstName!} ${user.lastName!}`,
+            lastName: `${user.lastName!} ${user.firstName!}`,
+          });
+        });
+        loading(false);
+      }
+    }
+
+    if(this.selectSearchType === SEARCH_OPTION_TYPE.LAST_NAME) {
+      const lastNameUserList = await CamundaRest.getUsersByLastNameGroups(this.token, this.bpmApiUrl, search, reviewerGroup);
```

Comment:
here, what if this api call fails? the loading will be infinite right? as in the case of the other 2 conditions?

Reply:
Thanks for thinking through all these situations 👍

Reply:
Addressed this scenario by setting loading - false at end of condition